4iprocessing/generate_initial_config.py

```diff
@@ -0,0 +1,130 @@
+parser.add_argument('--output_yaml_dir', type=str, required=True, help="final alignment outputs to specify in yaml file")
+
+
+def sort_rounds(rounds):
+    # function to sort rounds
+    Orginal_numbered_list = []
+    for i in range(len(rounds)):
+        int(re.search(r'\d+', rounds[i]).group())
+        Orginal_numbered_list.append(int(re.search(r'\d+', rounds[i]).group()))
+    sorted_list = sorted(Orginal_numbered_list)
+    final_sorted_list=[]
+    for num in sorted_list:
+        index = Orginal_numbered_list.index(num)
+        final_sorted_list.append(rounds[index])
+    #print(final_sorted_list)
+    return final_sorted_list
+
+
+if __name__ == '__main__':
+    args= parser.parse_args()
+
+    for bdir in args.input_dirs:
+        barcode = Path(Path(bdir)).name
+        print(barcode)
+        config ={}
+        config['Data']=[]
+        scope_list = [x for x in os.listdir(bdir) if 'ZSD' in x]
+        for scope in scope_list:
+            pdir = bdir + os.sep + scope
+            #round_list = [x for x in os.listdir(pdir) if bool(re.search('Time|Round [0-9]+(?!.)',x,re.IGNORECASE))&(Path(x).stem==Path(x).name)]
+            round_list = [x for x in os.listdir(pdir) if bool(re.search('Time|Round [0-9]+',x,re.IGNORECASE))&(Path(x).stem==Path(x).name)]
+            #round_list.sort(key=int)
+            round_list = sort_rounds(round_list)
+
+            print("round list is {}".format(round_list))
+
+            for round_num in round_list:
+                ppath = os.path.join(pdir, round_num)
+                czi_list = [x for x in os.listdir(ppath) if ('.czi' in x)&bool(re.search('20x',x,re.IGNORECASE))]
+                # print(czi_list)
+                # subd = {}
+                # subd['name'] = round_num
+                # subd['details'] = []
+
+                for czi_name in czi_list:
+                    fpath = os.path.join(ppath,czi_name)
+                    fpathr = fpath.replace(os.sep,'/')
+                    fpath2 ='"'+fpathr+'"'
+                    fpath2 = fpathr
+                    fpath = fpath2.replace("'",'')
+
+                    # each round of imaging is a multi-scene file like with dimensions STCZYX (only "time lapse rounds" have T>1). All rounds have S>1 and T>1
+                    # define the location of the reference channel in the list of channels
+                    # ideally the reference channel will be the image containing the nuclear fluorescence (or perhaps brightfield)
+                    # the reference image type (e.g. nuclear dye or brightfield) should be the consisently chosen for all rounds
+                    # if the image is a timelapse or round01 then choose the reference channel to be the last channel in the image set , =-1.
+
+                    ref_channel = -2
+                    if bool(re.search('time|1(?![0-9])',round_num, re.IGNORECASE)) and round_num!="Round 11": 
```

Comment:
need to make this catch any round greater than 10. right now it only looks for round 11 but in theory we may have 12 or 13 rounds in the future.

Reply:
Just confirming, here we only what this if statement to be executed if we are looking at the first round or the 20X timelapse.... is that correct?

Reply:
correct